### PR TITLE
log exception when unable to write to output

### PR DESCRIPTION
**Pull request by adyach**
## Description
Log exception when unable to write to output and change log4j props

### CODE REVIEW

**Comment by adyach** [comment]
👍

File: src/main/resources/log4j.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 log4j.rootCategory=INFO, CONSOLE
 
-LOG_PATTERN=[%d{yyyy-MM-dd HH:mm:ss.SSSXXX}] [%p] [%x] [%t] [%c] --- %m%n
+LOG_PATTERN=[%d{yyyy-MM-dd HH:mm:ss.SSSXXX}] [%p] [%x] [%t] [%c] --- %m %throwable{compact} %n
```
Comment (a1exsh) [inline src/main/resources/log4j.properties:3]: If this is supported by our log4j version, then by all means. :+1:

**Comment by adyach** [comment]
👍

File: app/src/main/resources/log4j.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 log4j.rootCategory=INFO, CONSOLE
 
-LOG_PATTERN=[%d{yyyy-MM-dd HH:mm:ss.SSSXXX}] [%p] [%x] [%t] [%c] --- %m%n
+LOG_PATTERN=[%d{yyyy-MM-dd HH:mm:ss.SSSXXX}] [%p] [%x] [%t] [%c] --- %m %throwable{compact} %n
```
Comment (a1exsh) [inline app/src/main/resources/log4j.properties:3]: Not sure about the extra space here:
```suggestion
LOG_PATTERN=[%d{yyyy-MM-dd HH:mm:ss.SSSXXX}] [%p] [%x] [%t] [%c] --- %m %throwable{compact}%n
```

**Comment by a1exsh** [comment]
:+1:

**Comment by adyach** [comment]
👍

File: app/src/main/resources/log4j.properties (hunk 2 above)
Comment (adyach) [reply app/src/main/resources/log4j.properties:3]: @a1exsh apologies, did not notice the suggestion

Comment (a1exsh) [reply app/src/main/resources/log4j.properties:3]: No worries :)

**Pull request merged**